client/src/modules/depots/modals/depot.modal.js

```diff
@@ -109,4 +129,39 @@ function DepotModalController($state, Depots, Notify, Session, params) {
       })
       .catch(Notify.handleError);
   }
+
+  /**
+   * destination depots handlers
+   */
+  function setNodeValue(childrens, depot) {
```

Comment:
Why not just use the TreeService to be this data structure?

That might give you better guarantees (if I recall, it is tested) than copy and pasting code from elsewhere.

Reply:
Alternatively, you could also use the `bhCheckboxTree` (https://github.com/IMA-WorldHealth/bhima/blob/master/client/src/js/components/bhCheckboxTree/bhCheckboxTree.js) to create the checkbox tree.  This would reduce the amount of logic in the controller.

Reply:
@jniles the `bhCheckboxTree` is a great tool and easy to use, however, there are some things that must be handled on it : 
- The selection of parent implies the selection of children (we cannot select only a parent without selecting children)

<img width="233" alt="image" src="https://user-images.githubusercontent.com/5445251/188398740-1ad60799-b511-4315-a65f-976f0ccb78a2.png">

- The component does not have a filter

I will add these points as issues but for now I will merge the current progress.

Reply:
I also think that we can write another issue to solve this problem, but also using the **bhCheckboxTree** here would also imply that we can modify the module for assigning deposits to users